plugins/explore/src/components/GroupsExplorerContent/GroupsDiagram.tsx

```diff
@@ -142,6 +147,7 @@ export function GroupsDiagram() {
       id: stringifyEntityRef(catalogItem),
       kind: catalogItem.kind,
       name: formatEntityRefTitle(catalogItem, { defaultKind: 'Group' }),
+      profile: catalogItem.spec?.profile,
```

Comment:
No need to introduce this new prop - it's probably best to just keep all computation in here, and leave the graph drawing "dumb".

`name: catalogItem.spec?.profile?.displayName || formatEntityRefTitle(catalogItem, { defaultKind: 'Group' })`

if you want to have some extra type safety, you can change the catalog call above to `catalogApi.getEntities<GroupEntity>` (with the type being from catalog-model). Then you'll get the right `spec` without worrying about `any` casts etc

Reply:
In a future update I'd like it if things that had a display name showed BOTH that and the "technical" name, where the latter maybe would be in small, muted text below the name or something. But not this PR

Reply:
is it by importing GroupEntity from '@backstage/catalog-model'?

Reply:
showing display name and metadata name should be a good idea,  maybe I can pick up that issue if there is any later, after fixing this PR haha

Reply:
yeah, that's the right package

Reply:
<GroupEntity> using this tag makes react think that is JSX, I cast it inside name object instead, what do you think?
`name:
        (catalogItem as GroupEntity).spec?.profile?.displayName ||
        formatEntityRefTitle(catalogItem, { defaultKind: 'Group' }),`

Reply:
![image](https://user-images.githubusercontent.com/33563774/123803562-9a875380-d916-11eb-88ff-db4d4695224c.png)
If I didn't cast it, the linter will show this error `Property 'displayName' does not exist on type 'string | number | boolean | JsonObject | JsonArray'.
  Property 'displayName' does not exist on type 'string'.ts(2339)`

Reply:
yeah i see now that i never got around to adding that type argument to the method ... sorry for lying, i really thought it was there :)